src/stress.codegen/TestAssemblyLoader.cs

```diff
@@ -111,21 +112,28 @@ private static string FindReferenceAssemblyInPaths(string assmName, IEnumerable<
             return assmPath;
         }
 
-        public UnitTestInfo[] GetTests<TDiscoverer>()
+        public string GetTests<TDiscoverer>()
             where TDiscoverer : ITestDiscoverer, new()
         {
             try
             {
                 var discoverer = new TDiscoverer();
 
-                return discoverer.GetTests(_assembly);
+                var tests = discoverer.GetTests(_assembly);
+
+                if (tests.Length > 0)
+                {
+                    CodeGenOutput.Info($"{_assembly.Assembly.FullName}: {tests.Length} tests discovered from assembly");
+                }
+
+                return JsonConvert.SerializeObject(tests);
```

Comment:
Serializing in to JSON allows to us to remove the crutch where we can't use certain types because the runtime doesn't know how to serialize/marshall it. So here we say "Okay CLR serializier, just marshall this string (wink) and we'll be happy."